platform/darwin/src/MGLStyle.h

```diff
@@ -200,6 +200,9 @@ static const NSInteger MGLStyleDefaultVersion = 9;
 /**
  Adds a new source to the current style.
  
+ @note Adding the same source instance more than once or reusing source 
```

Comment:
We could say which exception name gets raised under which circumstance, so the developer knows what to look for. For now, spelling out the exception name would only make sense for the exceptions we’ve given CamelCase names to.